### ISSUE TITLE
Failed to compile shader on windows (Intel HD Graphics)

### ISSUE THREAD

**Issue by fryeb** [servo/webrender#1251]
`ps_split_composite` does not compile. This causes the basic, scrolling and yuv examples to fail. The blob example continues to work. Perhaps a driver bug?

**Comment by kvark** [comment]
@fryeb could you provide the console output?

**Comment by fryeb** [comment]
```
C:\Users\fryeb\Desktop\webrender\webrender>cargo run --example basic
    Finished dev [unoptimized + debuginfo] target(s) in 0.4 secs
     Running `C:\Users\fryeb\Desktop\webrender\target\debug\examples\basic.exe`
OpenGL version 3.2.0 - Build 20.19.15.4331
Shader resource path: None
Failed to compile shader: "ps_split_composite"
ERROR: 0:22: 'constructor' :  cannot convert parameter 1 from 'float' to 'float'
ERROR: 0:18: 'return' : function return is not matching type:


thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Shader(Compilation("ps_split_composite", "ERROR: 0:22: \'constructor\' :  cannot convert parameter 1 from \'float\' to \'float\'\nERROR: 0:18: \'return\' : function return is not matching type: \n\n"))', src\libcore\result.rs:859
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: process didn't exit successfully: `C:\Users\fryeb\Desktop\webrender\target\debug\examples\basic.exe` (exit code: 101)
```

**Comment by kvark** [comment]
Thanks @fryeb ! The error is interesting ("cannot convert parameter 1 from 'float' to 'float'")... I can tweak the code a bit to see how we can work around this (seemingly a driver bug?), and we'll attempt to repro it internally.

**Comment by kvark** [comment]
If you can test a potential fix, here is a diff:
```diff
diff --git a/webrender/res/ps_split_composite.vs.glsl b/webrender/res/ps_split_composite.vs.glsl
index 20cf9a1..309d2c3 100644
--- a/webrender/res/ps_split_composite.vs.glsl
+++ b/webrender/res/ps_split_composite.vs.glsl
@@ -16,10 +16,12 @@ SplitGeometry fetch_split_geometry(int index) {
     vec4 data1 = texelFetchOffset(sSplitGeometry, uv, 0, ivec2(1, 0));
     vec4 data2 = texelFetchOffset(sSplitGeometry, uv, 0, ivec2(2, 0));
 
-    return SplitGeometry(vec3[4](
-        data0.xyz, vec3(data0.w, data1.xy),
-        vec3(data1.zw, data2.x), data2.yzw
-    ));
+    SplitGeometry geo;
+    geo.points[0] = data0.xyz;
+    geo.points[1] = vec3(data0.w, data1.xy);
+    geo.points[2] = vec3(data1.zw, data2.x);
+    geo.points[3] = data2.yzw;
+    return geo;
 }
 
 vec3 bilerp(vec3 a, vec3 b, vec3 c, vec3 d, float s, float t) {
```

**Comment by fryeb** [comment]
@kvark I have tested your solution and it works with all of the tests on my machine. I have also managed to get the following to work.
```glsl
SplitGeometry geo;
geo.points = vec3[4](
    data0.xyz, vec3(data0.w, data1.xy),
    vec3(data1.zw, data2.x), data2.yzw
);

return geo;
```
I think the fact that both versions work, while the current one does not suggests that it is probably a driver bug. Thank you for all your help @kvark .

**Comment by kvark** [comment]
Great, @fryeb , thanks for confirmation!
I've added the issue to https://github.com/servo/webrender/wiki/Driver-issues#1251---returning-with-struct-constructor
Would you mind filing a PR with a fix?

**Comment by fryeb** [comment]
@kvark I have submitted a pull request [here](https://github.com/servo/webrender/pull/1263)

**Comment by kvark** [comment]
Fixed by #1263
